controllers/core/pod_controller.go

```diff
@@ -91,21 +93,30 @@ func (r *PodReconciler) Reconcile(request custom.Request) (ctrl.Result, error) {
 	// has initialized (or it will be stuck till the next re-sync period or Pod update).
 	// Once the Pod has been initialized if it's managed then wait till the asynchronous
 	// operation on the Node has been performed and node is ready to server requests.
-	node, found := r.NodeManager.GetNode(pod.Spec.NodeName)
-	if !found {
-		logger.V(1).Info("pod's node is not yet initialized by the manager, will retry")
-		return PodRequeueRequest, nil
-	} else if !node.IsManaged() {
-		logger.V(1).Info("pod's node is not managed, skipping pod event")
+	node, foundInCache := r.NodeManager.GetNode(pod.Spec.NodeName)
+
+	nodeDeletedInCluster := false
+	if !isDeleteEvent && r.isNodeExistingInCluster(pod, logger) {
+		if !foundInCache {
+			logger.V(1).Info("pod's node is not yet initialized by the manager, will retry", "Requested", request.NamespacedName.String(), "Cached pod name", pod.ObjectMeta.Name, "Cached pod namespace", pod.ObjectMeta.Namespace)
+			return PodRequeueRequest, nil
+		} else if !node.IsManaged() {
+			logger.V(1).Info("pod's node is not managed, skipping pod event", "Requested", request.NamespacedName.String(), "Cached pod name", pod.ObjectMeta.Name, "Cached pod namespace", pod.ObjectMeta.Namespace)
+			return ctrl.Result{}, nil
+		} else if !node.IsReady() {
+			logger.V(1).Info("pod's node is not ready to handle request yet, will retry", "Requested", request.NamespacedName.String(), "Cached pod name", pod.ObjectMeta.Name, "Cached pod namespace", pod.ObjectMeta.Namespace)
+			return PodRequeueRequest, nil
+		}
+	} else if foundInCache && !node.IsManaged() {
```

Comment:
If node is no longer in cluster, the node reconciler will update node cache and remove the node. We don't care the pod which is not using the SGP feature and its node is not managed. Calling `isManaged()` is fine since the API just check if the node is an initialized node when the node is in our cache.